sips/cluster_consensus.md

```diff
@@ -0,0 +1,127 @@
+For example, take a look at the `AttestationData` type.
+```go
+type AttestationData struct {
+	Slot            Slot
+	Index           CommitteeIndex
+	BeaconBlockRoot Root `ssz-size:"32"`
+	Source          *Checkpoint
+	Target          *Checkpoint
+}
+```
+The only validator-dependent field is `CommitteeIndex` and it does not have to be agreed on.
+
+For the `Sync Committee` duty, operators agree on a `phase0.Root` data which is also independent of the validator.
+
+On the other hand, a post-consensus phase is still required for each duty that was previously aggregated. Thus, the partial signatures for each validator must still be shared between parties.
+
+## Improvement
+
+This proposal helps to decrease the number of messages exchanged in the network and the processing cost.
+
+According to Monte Carlo simulations using a dataset based on the Mainnet, this proposal reduces to $21.60$% the current number of messages exchanged in the network. Note that this result includes aggregating the post-consensus messages into a single message.
+
+Regarding the number of bits exchanged, we estimate that this proposal will reduce the current value to, at least, $52.96$%. Notice that this reduction is not as significant as the number of messages reduction due to the larger post-consensus messages.
+
+Again with Monte Carlo simulations using the Mainnet dataset, the number of attestation duties aggregated presented the following distribution (notice that it also represents the number of partial signature messages merged into a single message).
+
+<p align="center">
+<img src="./images/cluster_consensus/aggregated_duties.png"  width="50%" height="10%">
+</p>
+
+## Spec changes
+
+### Design
+
+Currently, an operator manages many `Validator` objects. Each has a `DutyRunner` for a duty type, each with its own `QBFTController` with its unique ID (that is also inserted in its associated messages).
+
+<p align="center">
+<img src="./images/cluster_consensus/previous_scheme.drawio.png"  width="50%" height="10%">
+</p>
+
+For the proposed change to take place, different `Validator` objects should use the same `QBFT Instance`. For that, we propose decoupling the `QBFTController` object from the `DutyRunner`.
+
+<p align="center">
+<img src="./images/cluster_consensus/new_scheme.drawio.png"  width="50%" height="10%">
+</p>
+
+### New IDs
+
+Since a single `QBFT instance` will be responsible for several validator duties, its ID must not be dependent on a validator key but rather on a cluster of operators. For that, we propose changing the `MessageID` from
+
+```mermaid
+flowchart RL
+	subgraph MessageID
+		Domain
+		ValidatorPublicKey
+		Role
+	end
+```
+
+to
+
+
+```mermaid
+flowchart RL
+	subgraph MessageID
+		Domain
+		OperatorCluster
+		Role
+	end
+```
+
+### QBFT Controller
+
+The current `QBFTController` structure allows only one consensus instance at a time. This must change, extending the `QBFTController` into a router of messages for the different instances.
```

Comment:
In my opinion we can simply drop `QbftController` from spec
It used to be there when a runner ran several instances at the same time...

We don't do this anymore.
Even if it is needed by the implementation people I don't see why spec needs this

Reply:
Specifically for this change, I think we should keep it to handle the several operator's QBFT instances running in parallel.